src/crypto/crypto_keys.cc

```diff
@@ -1237,6 +1239,45 @@ void KeyObjectHandle::GetAsymmetricKeyType(
   args.GetReturnValue().Set(key->GetAsymmetricKeyType());
 }
 
+bool KeyObjectHandle::CheckKeyData() const {
+  MarkPopErrorOnReturn mark_pop_error_on_return;
+
+  bool isValid;
+  const ManagedEVPPKey& key = data_->GetAsymmetricKey();
+  EVPKeyCtxPointer ctx(EVP_PKEY_CTX_new(key.get(), nullptr));
+  switch (EVP_PKEY_id(key.get())) {
+    case EVP_PKEY_EC: {
```

Comment:
Since calling this function on non-EC keys leads to `UNREACHABLE` anyway, could you replace this with a `CHECK_EQ(EVP_PKEY_id(key.get()), EVP_PKEY_EC)` and rename the function to something like `CheckEcKeyData`?